base/abstractarraymath.jl

```diff
@@ -287,6 +307,71 @@ function repeat(a::AbstractVecOrMat, m::Integer, n::Integer=1)
     return b
 end
 
+function repeat(a::AbstractVecOrMat, m::AbstractVector{<:Integer}, n::Integer=1)
+    o, p = size(a,1), size(a,2)
+    length(m) == o || throw(ArgumentError("m must be a single Integer or an AbstractVector{<:Integer} of same dimension as a"))
+    b = similar(a, sum(m), p*n)
+    for j=1:n
+        d = (j-1)*p+1
+        R = d:d+p-1
+        c = 1
+        for (i,r) ∈ enumerate(m)
+            v = a[i,:]
+            while r > 0
+                b[c, R] = v
+                c += 1
+                r -= 1
+            end
+        end
+    end
+    return b
+end
+
+function repeat(a::AbstractVecOrMat, m::Integer, n::AbstractVector{<:Integer})
+    o, p = size(a,1), size(a,2)
+    length(n) == p || throw(ArgumentError("n must be a single Integer or an AbstractVector{<:Integer} of same dimension as a"))
+    b = similar(a, o*m, sum(n))
+    sumj = zero(eltype(n))
+    for (j, r) ∈ enumerate(n)
+        v = @view a[:,j]
+        d = sumj + 1
+        sumj += r
+        R = d:sumj
```

Comment:
No need to store the range in `R`.

Reply:
You're right, must be a remnant from an earlier implementation. Fixed.